tests/integration/pilot/mcs/discoverability/discoverability_test.go

```diff
@@ -234,13 +236,77 @@ func checkClustersReached(t framework.TestContext, src, dest echo.Instance, clus
 		dest.Config().Service,
 		dest.Config().Namespace.Name())
 
-	src.CallWithRetryOrFail(t, echo.CallOptions{
+	_, err := src.CallWithRetry(echo.CallOptions{
 		Address:   address,
 		Target:    dest,
 		Count:     50,
 		PortName:  "http",
 		Validator: echo.And(echo.ExpectOK(), echo.ExpectReachedClusters(clusters)),
 	}, retryTimeout)
+	if err != nil {
+		t.Fatal(fmt.Errorf("%v\nCluster Details:\n%s", err, getClusterDetailsYAML(t, src, dest)))
+	}
+}
+
+func getClusterDetailsYAML(t framework.TestContext, src, dest echo.Instance) string {
+	// Add details about the configuration to the error message.
+	type ClusterInfo struct {
+		Cluster            string   `json:"cluster"`
+		DestinationPodIPs  []string `json:"destinationPodIPs,omitempty"`
+		EastWestGatewayIPs []string `json:"eastWestGatewayIPs,omitempty"`
+	}
+
+	type Details struct {
+		SourceOutboundIPs []string      `json:"sourceOutboundIPs,omitempty"`
+		Clusters          []ClusterInfo `json:"clusters,omitempty"`
+	}
+	details := Details{}
+
+	destName := dest.Config().Service
+	destNS := dest.Config().Namespace.Name()
+	istioNS := istio.GetOrFail(t, t).Settings().SystemNamespace
+
+	for _, c := range t.Clusters() {
+		info := ClusterInfo{
+			Cluster: c.StableName(),
+		}
+
+		// Get pod IPs for service B.
+		pods, err := c.PodsForSelector(context.TODO(), destNS, "app="+destName)
+		if err == nil {
+			for _, destPod := range pods.Items {
+				info.DestinationPodIPs = append(info.DestinationPodIPs, destPod.Status.PodIP)
+			}
+		}
+
+		// Get the East-West Gateway IP
+		svc, err := c.Kube().CoreV1().Services(istioNS).Get(context.TODO(), "istio-eastwestgateway", kubeMeta.GetOptions{})
```

Comment:
You can get this with plenty of context using `istioctl x internal-debug networkz` or equiv GET on pilot.